cmd/dagger/flags.go

```diff
@@ -448,6 +449,12 @@ func (v *secretValue) Set(s string) error {
 		s = secretSource + "://" + val
 	}
 
+	const cacheKeyPrefix = ",cacheKey="
```

Comment:
Agreed, I don't love this syntax either. Dunno, I feel like maybe it might be worth struggling through the pain of getting secret provider as *full URIs* so this could work?

But no real other suggestions :cry:

Reply:
It's mainly the `cmd://echo blah blah?cacheKey=my-tokens` situation that's hard to deal with, since that's not a valid URI (it complains the host cannot have spaces, which makes sense).

I *did* just realize that `cmd:///echo blah blah?cacheKey=my-tokens` (3 `/` instead of 2) is valid because the command is technically part of the path rather than host and spaces get url encoded)

I wouldn't want users to have to type 3 `/`s though, could special case it I guess to automatically put an extra `/`. Also some potential weirdness about disambiguating `/command` from `command`, but getting outrageously obscure at that point.

I'll try something out quick here and see if it feels any better/worse than the current solution

Reply:
Got this working w/ a slightly different approach that required an extra chunk of careful query param parsing in cmd/dagger/flags.go, but added integ+unit test coverage to cover it